From ce71383e778692563785026bfcf1055ce787fb17 Mon Sep 17 00:00:00 2001
From: Victor Gu <xigu@marvell.com>
Date: Thu, 12 Nov 2015 19:00:58 +0800
Subject: [PATCH 0494/1240] armada3700: io-dec: disable print and improve print

1.Disable the print in io decode module by default.
2.Improve few prints

Change-Id: Ie80c69078e97b802077b64acb60b17bec7e00935
Signed-off-by: Victor Gu <xigu@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/24799
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 drivers/misc/mvebu_io_addr_dec.c | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/drivers/misc/mvebu_io_addr_dec.c b/drivers/misc/mvebu_io_addr_dec.c
index 527d856..2f8fd43 100644
--- a/drivers/misc/mvebu_io_addr_dec.c
+++ b/drivers/misc/mvebu_io_addr_dec.c
@@ -16,7 +16,7 @@
  * ***************************************************************************
  */
 
-#define DEBUG
+/* #define DEBUG */
 #include <common.h>
 #include <asm/arch-mvebu/mvebu.h>
 #include <asm/arch-mvebu/mbus.h>
@@ -81,7 +81,7 @@ int set_io_addr_dec(struct mbus_win_map *win_map, struct dec_win_config *dec_win
 
 		/* set base address*/
 		writel(base, MVEBU_DEC_WIN_BASE_REG(dec_win->dec_win_base, id, dec_win->win_offset));
-		/* set remap window */
+		/* set remap window, some unit does not have remap window */
 		if (id < dec_win->max_remap)
 			writel(base, MVEBU_DEC_WIN_REMAP_REG(dec_win->dec_win_base, id, dec_win->win_offset));
 		/* set control register */
@@ -90,10 +90,14 @@ int set_io_addr_dec(struct mbus_win_map *win_map, struct dec_win_config *dec_win
 		ctrl |= win->enabled << MVEBU_DEC_WIN_CTRL_EN_OFF;
 		writel(ctrl, MVEBU_DEC_WIN_CTRL_REG(dec_win->dec_win_base, id, dec_win->win_offset));
 
-		debug("set_io_addr_dec %d ctrl(0x%x) base(0x%x) remap(%x)\n",
+		debug("set_io_addr_dec %d result: ctrl(0x%x) base(0x%x)",
 		      id, readl(MVEBU_DEC_WIN_CTRL_REG(dec_win->dec_win_base, id, dec_win->win_offset)),
-		      readl(MVEBU_DEC_WIN_BASE_REG(dec_win->dec_win_base, id, dec_win->win_offset)),
-		      readl(MVEBU_DEC_WIN_REMAP_REG(dec_win->dec_win_base, id, dec_win->win_offset)));
+		      readl(MVEBU_DEC_WIN_BASE_REG(dec_win->dec_win_base, id, dec_win->win_offset)));
+		if (id < dec_win->max_remap)
+			debug(" remap(%x)\n",
+			      readl(MVEBU_DEC_WIN_REMAP_REG(dec_win->dec_win_base, id, dec_win->win_offset)));
+		else
+			debug("\n");
 	}
 	return 0;
 }
-- 
1.9.1

